File: BackEnd/FastApiServer/models/vocabulary.py
# models/vocabulary.py
from sqlalchemy import Column, BigInteger, String
from database import Base

class Vocabulary(Base):
    __tablename__ = "vocabulary"

    vocabulary_id = Column(BigInteger, primary_key=True, index=True)
    japanese = Column(String(255), nullable=False)
    korean = Column(String(255), nullable=False)
    japanese_read = Column(String(255), nullable=False)
    type = Column(String(30), nullable=True)

File: BackEnd/SpringBootServer/src/main/java/com/ssafy/backend/domain/vocabulary/entity/Vocabulary.java
package com.ssafy.backend.domain.vocabulary.entity;

import com.ssafy.backend.global.common.entity.BaseEntity;
import jakarta.persistence.*;
import lombok.*;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PROTECTED)
@Builder
@Entity
public class Vocabulary extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(columnDefinition = "INT UNSIGNED")
    private Long id;

    @Column(nullable = false, columnDefinition = "VARCHAR(50)")
    private String japanese;

    @Column(nullable = false, columnDefinition = "VARCHAR(200)")
    private String korean;

    @Column(nullable = false, columnDefinition = "VARCHAR(50)")
    private String japaneseRead;

    @Column(columnDefinition = "VARCHAR(50)")
    private String type;
}
